project1/app/Models/ProductCategory.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class ProductCategory extends Model
+{
+    use HasFactory;
+    protected $table = "product_category";
+    protected $guarded =[];
+
+    public function category(){
+        return $this->belongsTo(Category::class);
+    }
+
+    public function product(){
+        return $this->belongsTo(Product::class);
+    }  
+
+    public static function store($request, $insert){
+        $insert1 = null;
+        if(isset($insert)){
+            $params[0] = ['category_id' => $request->category_id,'product_id' => $insert->id];
+            $insert1 = ProductCategory::create($params[0]);
+            if($request->category_id_1 != null){
+                $params[1] = ['category_id' => $request->category_id_1,'product_id' => $insert->id];
+                $insert1 = ProductCategory::create($params[1]);
+                if($request->category_id_2 != null){
+                    $params[2] = ['category_id' => $request->category_id_2,'product_id' => $insert->id];
+                    $insert1 = ProductCategory::create($params[2]);
+                }
+            }
+        }
+        return $insert1;
+    }
+
+    public static function ProductCategoryUpdate($params,$productCategory,$update){
+        
+        if($params['category_id'] == null){  
+            if(!empty($productCategory)){
+                $update = $productCategory->delete();
+            }
+        }else{   
+            if(!empty($productCategory)){
+                $update = $productCategory->update($params);
+            }else{
+                ProductCategory::create($params);  
+            }
+        }return $update;
+    }
+
+    public static function updateProductCategory($request, $id, $update){
+        $productCategory = ProductCategory::where('product_id','=',$id)->get();
+        $params[2] = ['category_id' => $request->category_id_2,'product_id' => $id]; 
+        $params[1] = ['category_id' => $request->category_id_1,'product_id' => $id]; 
+        $params[0] = ['category_id' => $request->category_id,'product_id' => $id];
```

Comment:
Mô tả cách làm category này giúp a xem

Reply:
Categories của em có 3 cấp:
Ví dụ: Linh kiện 3 cấp: Linh kiện ->loại linh kiện -> nhãn hàng
          Điện thoại 2 cấp: Điện thoại -> nhãn hàng
          Khác có 1 cấp
Ở đây của em product có thể quan hệ tối đa 3 categories
em lấy $productCategory có tối đa 3 giá trị
Nên em chạy lệnh for từ 2 đến 0
{ nếu category_id == null, 
nếu tồn tại productCategory em sẽ xóa không thì bỏ qua
category_id khác null 
em sẽ cập nhật hoặc thêm nếu productCategory có tồn tại hay không  }

Reply:
Nếu anh muốn nhiều cấp hơn nữa thì sao

Reply:
Phải nâng cấp quản lý categories lên anh.
Trong trường hợp hàm này thì ta có thể chạy lệnh foreach từ (mảng category_id) trong đó $i = 0 và tăng lên qua từng vòng lặp.
Rồi ta sử dụng $i đó cho lệnh for